src/components/layout/Layout.tsx

```diff
@@ -4,24 +4,27 @@ import { Outlet } from 'react-router-dom';
 import { getUser } from '../../util/api';
 import Header from './header/Header';
 import Footer from './footer/Footer';
+import { User } from '../../types/types';
 
 function Layout() {
-  const [user, setUser] = useState(null);
+  const [user, setUser] = useState<User>();
 
-  const handleUserLoad = async () => {
-    try {
-      const { data } = await getUser();
+  useEffect(() => {
+    const handleUserLoad = async () => {
+      try {
+        const user = await getUser();
 
-      setUser(data[0]);
-    } catch (err) {
-      console.error(err.message);
-    }
-  };
+        setUser(user);
+      } catch (err) {
+        // TODO: err.message 하고 싶었음
+        console.error(err);
```

Comment:
```suggestion
        const error = err as Error
        console.error(error.message);
```

이렇게 재할당을 통해 해결할 수 있습니다.